app/models/vm_cloud_reconfigure_task.rb

```diff
@@ -0,0 +1,49 @@
+class VmCloudReconfigureTask < MiqRequestTask
+  alias_attribute :vm, :source
+
+  validate :validate_request_type, :validate_state
+
+  AUTOMATE_DRIVES = false
+
+  def self.base_model
+    VmCloudReconfigureTask
+  end
+
+  def self.get_description(req_obj)
+    name = nil
+    if req_obj.source.nil?
+      # Single source has not been selected yet
+      if req_obj.options[:src_ids].length == 1
+        v = Vm.find_by(:id => req_obj.options[:src_ids].first)
+        name = v.nil? ? "" : v.name
+      else
+        name = "Multiple VMs"
+      end
+    else
+      name = req_obj.source.name
+    end
+    flavor = Flavor.find_by(:id => req_obj.options[:instance_type].first)
+    flavor_name = flavor.nil? ? "" : flavor.name
```

Comment:
@gmcculloug On the question of checking for nil vs. raising an exception, we've got 2 places where I"m doing this here. This one is for get_description. This is a method that shouldn't raise an exception if it can't find a flavor, so I think what's happening here is correct. For the second one, in the do_request call, if the flavor isn't found, then we probably should raise an exception -- I'll modify the second one to raise an exception if flavor isn't found in the find.